### ISSUE TITLE
Issue with CatalogTyped constants

### ISSUE THREAD

**Issue by gabizou** [SpongePowered/SpongeAPI#883]
In general, most CatalogTypes are not registered or haven't completed registration at the latest by the `GamePostInitializationEvent`. In layman's terms, this means that if any use of some `CatalogType`s are used (even with relation to the Data API), the types themselves may still result in `null`s. And with the recent addition of `DataConstants` in SpongeCommon, this is going to be all the more important considering there are a few areas where constants are absolutely required to not be `null`.

The biggest issue is when Data is involved. Since already made self evident with some of the `DataManipulator` tests, if the `CatalogType`s have not been registered by the time the test is run, things break down due to non-null validation checks. 

Ultimately, this is an issue of the `GameRegistry` and how the registry is actually handled.

**Comment by simon816** [comment]
Isn't this more of an issue with how SpongeCommon handles the registry? Seems like https://github.com/SpongePowered/SpongeCommon/issues/6 is related

**Comment by gabizou** [comment]
> Isn't this more of an issue with how SpongeCommon handles the registry?



> Ultimately, this is an issue of the GameRegistry and how the registry is actually handled.

**Comment by Zidane** [comment]
Fixed with @gabizou 's new registry system.
